src/encode/json.rs

```diff
@@ -162,6 +171,34 @@ impl Deserialize for JsonEncoderDeserializer {
         Ok(Box::<JsonEncoder>::default())
     }
 }
+#[cfg(feature = "log_kv")]
+mod kv {
+    use log::kv::VisitSource;
+    use std::collections::BTreeMap;
+
+    pub(crate) type Map = BTreeMap<String, String>;
+
+    pub(crate) fn get_attributes(source: &dyn log::kv::Source) -> anyhow::Result<Map> {
+        struct Visitor {
+            inner: Map,
+        }
+        impl<'kvs> VisitSource<'kvs> for Visitor {
+            fn visit_pair(
+                &mut self,
+                key: log::kv::Key<'kvs>,
+                value: log::kv::Value<'kvs>,
+            ) -> Result<(), log::kv::Error> {
+                self.inner.insert(format!("{key}"), format!("{value}"));
+                Ok(())
+            }
+        }
```

Comment:
Why have the struct and impl defined within a function?

Reply:
My reasoning was that they are implementation details and do not serve a purpose outside of this function.

Reply:
@ellttBen Please move the impl out of the fn definition into the `kv` namespace.